napari/layers/utils/layer_utils.py

```diff
@@ -62,6 +62,26 @@ def _inner(func):
     return _inner
 
 
+def _nanmin(array):
+    """
+    call np.min but fall back to avoid nan and inf if necessary
+    """
+    min = array.min()
+    if not np.isfinite(min):
+        min = np.ma.masked_invalid(array).min()
```

Comment:
```suggestion
        min = np.min(np.ma.masked_invalid(array))
```

As above

Reply:
Though the use of `np.ma` might mean that we don't support other arrays anyway... Though I appreciate learning about a new function 😊 I might suggest using the long form:

```suggestion
        min = np.min(array[np.isfinite(array)])
```

Thoughts?

Reply:
I took a quick look, and things seem to work fine with dask (and tensorstore I think, since tests passed). however, the explicit form seems a bit faster, on top of being clearer. I'll go with that!

```python
a = np.random.rand(100, 100, 100)
a[:20] = np.nan
a[80:] = np.inf

%timeit np.max(np.ma.masked_invalid(a))
7.57 ms ± 23.2 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
%timeit np.max(a[np.isfinite(a)])
1.7 ms ± 5.76 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

a_dask = da.array(a)

%timeit np.max(np.ma.masked_invalid(a_dask))
8.23 ms ± 32.2 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
%timeit np.max(a_dask[np.isfinite(a_dask)])
2.05 ms ± 229 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

Reply:
Actually... we might be making things worse:

```python
In [33]: %timeit np.min(a)
293 µs ± 675 ns per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [34]: %timeit np.nanmin(a)
785 µs ± 4.13 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [35]: %timeit _nanmin(a)
1.96 ms ± 8.46 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```